x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/aggregate/Sum.java

```diff
@@ -53,12 +64,16 @@ public class Sum extends NumericAggregate implements SurrogateExpression {
                 tag = "docsStatsSumNestedExpression"
             ) }
     )
-    public Sum(Source source, @Param(name = "number", type = { "double", "integer", "long" }) Expression field) {
-        this(source, field, Literal.TRUE);
+    public Sum(
+        Source source,
+        @Param(name = "number", type = { "double", "integer", "long" }) Expression field,
+        Configuration configuration
+    ) {
+        this(source, field, Literal.TRUE, configuration);
     }
 
-    public Sum(Source source, Expression field, Expression filter) {
-        super(source, field, filter, emptyList());
+    public Sum(Source source, Expression field, Expression filter, Configuration configuration) {
+        super(source, field, filter, emptyList(), configuration);
```

Comment:
I was thinking you'd read the value from the configuration in the ctor rather than keep the configuration around. Heck - in this public ctor you could just pass the `boolean` and read if from the configuration in the one above. That way you could call this one in tests if you wanted.

Reply:
I was planning to do that, but as scalar functions already have this configuration stored, I thought it was better to keep the same mechanism (?)